modules/sdk-core/src/bitgo/utils/tss/ecdsa/ecdsa.ts

```diff
@@ -728,11 +960,11 @@ export class EcdsaUtils extends baseTSSUtils<KeyShare> {
       txRequest.txRequestId,
       requestType,
       SendShareType.MUShare,
-      { muShare, dShare, i: muShare.i }
+      step2Return.muDShare
     )) as DShare;
 
     const userSShare = await ECDSAMethods.createUserSignatureShare(
-      userOmicronAndDeltaShare.oShare,
+      step2Return.oShare as OShare,
```

Comment:
Why are we casting?

Reply:
good catch, my bad it's not needed here

Reply:
in the case of the hot wallet flow this step2Return would hold the actual OShare which is an OShare object, in the case of external signer it would return an encryptedOShare as a string. This is explained as a comment on the type definition of step2Return.

Reply:
We can have better types instead of casting. Casting is almost never the fix for production app code.